interface/resources/qml/hifi/avatarapp/AvatarsModel.qml

```diff
@@ -20,18 +20,48 @@ ListModel {
             .split('%marketId%').join(extractMarketId(avatarUrl));
 
         return marketItemUrl;
+    }
+	
+	function makeMarketThumbnailUrl(marketId) {
+        var avatarThumbnailUrl = "https://hifi-metaverse.s3-us-west-1.amazonaws.com/marketplace/previews/%marketId%/large/hifi-mp-%marketId%.jpg"
+            .split('%marketId%').join(marketId);
+            
+        return avatarThumbnailUrl;
+    }
+	
+    function trimFileExtension(url) {
+        url = url.substring(0, (url.indexOf("#") == -1) ? url.length : url.indexOf("#"));
+        url = url.substring(0, (url.indexOf("?") == -1) ? url.length : url.indexOf("?"));
+        url = url.substring(0, url.lastIndexOf("."));
+
+        return url;
+    }
+	
+	function imageExists(image_url){
+
+        var http = new XMLHttpRequest();
+
+        http.open('HEAD', image_url, false);
+        http.send();
+
+        return http.status != 404;
+
     }
 
     function makeThumbnailUrl(avatarUrl) {
         var marketId = extractMarketId(avatarUrl);
-        if (marketId === '') {
+        if (marketId !== '') {
+            return makeMarketThumbnailUrl(marketId);
+        }
+        
+        var avatarThumbnailFileUrl = trimFileExtension(avatarUrl)+".jpg";
```

Comment:
it would be nice if this came from the fst (maybe if it’s not in there, it falls back to this)

this also might affect baking...I think with this solution, the thumbnails won’t be transferred over when you bake, so you’ll lose the thumbnail

Reply:
Do we have a method to bake avatars in interface? if you are using the oven it still has files. it technically does not care what your avatar extension is it just cares that a jpg has the same name. this method will work even if you just wear avatars as a gltf or a fbx.

Reply:
Related: https://github.com/kasenvr/project-athena/issues/84

Reply:
sure but when you bake using the oven, it copies your files over to a new directory.  will this thumbnail image also get copied to the new location?

Reply:
space missing around the + operator `trimFileExtension(avatarUrl) + ".jpg";`